modules/planning/tasks/deciders/st_bounds_decider/st_obstacles_processor.cc

```diff
@@ -171,5 +212,27 @@ bool STObstaclesProcessor::IsPathPointAwayFromObstacle(
   return true;
 }
 
+bool STObstaclesProcessor::IsADCOverlappingWithObstacle(
+    const PathPoint& adc_path_point, const Box2d& obs_box,
+    const double l_buffer) const {
+  // Convert reference point from center of rear axis to center of ADC.
```

Comment:
Would use the `ComputeCOMPosition` in `vehicle_state_provider` and convert to SL frame make the vehicle shape more precise?

Reply:
I checked that class. It seems that is providing the adc's current state based on localization and chassis. However, here I'm trying to calculate a possible collision for some futuristic position :)